src/shims/unix/env.rs

```diff
@@ -276,7 +276,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
     fn linux_gettid(&mut self) -> InterpResult<'tcx, Scalar> {
         let this = self.eval_context_ref();
-        this.assert_target_os("linux", "gettid");
+        this.assert_target_os_is_one_of(&["linux", "android"], "gettid");
```

Comment:
This diff seems unnecessary? `linux_gettid` is not called from any new places in this PR, is it?

Reply:
Yes, not anymore. Will remove.